pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -416,12 +416,8 @@ func (e *Controller) syncService(ctx context.Context, key string) error {
 	var totalNotReadyEps int
 
 	for _, pod := range pods {
-		if len(pod.Status.PodIP) == 0 {
-			klog.V(5).Infof("Failed to find an IP for pod %s/%s", pod.Namespace, pod.Name)
-			continue
-		}
-		if !tolerateUnreadyEndpoints && pod.DeletionTimestamp != nil {
-			klog.V(5).Infof("Pod is being deleted %s/%s", pod.Namespace, pod.Name)
+		if !endpointutil.ShouldPodBeInEndpointSlice(pod, tolerateUnreadyEndpoints) {
```

Comment:
At first glance, this looked wrong to me, but I believe it is correct. The second param here is used as `includeTerminating` in the function that's being called. Here's what we're doing for EndpointSlice:

https://github.com/kubernetes/kubernetes/blob/114cdea709b7046c439d9c4088cb9ab75105a0f0/pkg/controller/endpointslice/reconciler.go#L156

We definitely do not want to consider the feature gate here since Endpoints don't have a terminating condition, so I think the logic is correct here. With that said, it may be slightly easier to parse if we just used the value of the service field directly here (`service.Spec.PublishNotReadyAddresses`) instead of setting it to the `tolerateUnreadyEndpoints` var first.

Reply:
this comes from line 406
```
	// If the user specified the older (deprecated) annotation, we have to respect it.
	tolerateUnreadyEndpoints := service.Spec.PublishNotReadyAddresses
```

but I can change it to use the service field directly

Reply:
yeah that's what I meant - would just make it a bit more straightforward when comparing usage of ShouldPodBeInEndpointSlice across controllers.

Reply:
added a comment to make it clear the difference between controllers

Reply:
Why even use a separate var? It doesn't look like `tolerateUnreadyEndpoints` is used anywhere else, can we just use `service.Spec.PublishNotReadyAddresses`?

Reply:
done